contracts/Colony.sol

```diff
@@ -285,8 +285,24 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
   }
 
   function finishUpgrade2To3() public always {
-    ColonyAuthority(address(authority)).setRoleCapability(
-      uint8(ColonyDataTypes.ColonyRole.Root),
+    ColonyAuthority colonyAuthority = ColonyAuthority(address(authority));
+
+    colonyAuthority.setRoleCapability(
+     uint8(ColonyDataTypes.ColonyRole.ArchitectureSubdomain),
+      address(this),
+      bytes4(keccak256("setArbitrationRole(uint256,uint256,address,uint256,bool)")),
+      true
+    );
+
+    colonyAuthority.setRoleCapability(
+     uint8(ColonyDataTypes.ColonyRole.Root),
+      address(this),
+      bytes4(keccak256("setArbitrationRole(uint256,uint256,address,uint256,bool)")),
+      true
+    );
+
+    colonyAuthority.setRoleCapability(
+     uint8(ColonyDataTypes.ColonyRole.Root),
```

Comment:
Interesting -- is the idea here that every version of colony contains it's own custom upgrade function? Seems reasonable if a colony must be upgraded in sequence (kind of like a migration) -- what sorts of things will end up here? Looks like just authority changes?

I'm  assuming that the next release will delete this function and introduce a `finishUpgrade3to4`? Also, why isn't this function called in the `finishUpgrade` function? Are colonies expected to call this manually?

Reply:
This upgrade function was not introduced here. Please read PR https://github.com/JoinColony/colonyNetwork/pull/667 and discussions below it to understand the upgrade process we went for there.

Reply:
Ah I see -- the currently-deployed version do not have the `finishUpgrade` function, so we can't leverage it in this version. Ok makes sense.